lib/config/customer_io_config.dart

```diff
@@ -0,0 +1,53 @@
+import '../customer_io_enums.dart';
+import '../customer_io_plugin_version.dart' as plugin_info show version;
+import 'in_app_config.dart';
+import 'push_config.dart';
+
+class CustomerIOConfig {
+  final String source = 'Flutter';
+  final String version = plugin_info.version;
+
+  final String cdpApiKey;
+  final String? migrationSiteId;
+  final Region? region;
+  final CioLogLevel? logLevel;
+  final bool? autoTrackDeviceAttributes;
+  final String? apiHost;
+  final String? cdnHost;
+  final int? flushAt;
+  final int? flushInterval;
+  final InAppConfig? inAppConfig;
+  final PushConfig pushConfig;
+
+  CustomerIOConfig({
```

Comment:
i think `trackApplicationLifecycleEvents` is not here as its part of another ticket, correct?

Reply:
I focused mainly on Dart file changes here. native SDK implementation and testing will be handled in the other ticket you mentioned.